mono/utils/mono-threads-coop.c

```diff
@@ -16,6 +16,15 @@
 #define _DARWIN_C_SOURCE
 #endif
 
+/* On platforms that doesn't have full context support (or doesn't do conservative stack scan), use copy stack data */
+/* when entering safe/unsafe GC regions. For platforms with full context support (doing conservative stack scan), */
+/* there is already logic in place to take context before getting in a state where thread could be conservative */
+/* scanned by GC. Avoiding doing additional stack copy will increse performance when entering safe/unsafe regions */
+/* when running in hybrid/cooperative supspend mode. */
+#if defined (HOST_IOS) || defined (HOST_WATCHOS) || defined (HOST_TVOS) || defined (HOST_ANDROID) || defined (HOST_WASM)
```

Comment:
I'd like also `!defined (USE_COPY_STACK_DATA) &&` and then we should have a linux CI lane (perhaps the coop suspend lane) with `-DUSE_COPY_STACK_DATA` so that we get some coverage of the implementation.

Reply:
I could add that into the condition above and then set CFLAGS in run-jenkins.sh for selected CI configuration (like linux coop suspend CI lanes 32/64-bit).